core/test-output-stream.ts

```diff
@@ -106,12 +107,40 @@ export class TestOutputStream extends ReadableStream {
       return "undefined";
    }
 
-   private _getErrorYaml(error: MatchError): string {
-      return " ---\n" +
-      "   message: \"" + error.message.replace(/\\/g, "\\\\").replace(/"/g, "\\\"") + "\"\n" +
-      "   severity: fail\n" +
-      "   data:\n" +
-      "     got: " + JSON.stringify(error.actualValue) + "\n" +
-      "     expect: " + JSON.stringify(error.expectedValue) + "\n ...\n";
+   private _writeMatchErrorOutput(error: MatchError): void {
+
+       let sanitisedMessage = error.message.replace(/\\/g, "\\\\").replace(/"/g, "\\\"");
+       let sanitisedActual = JSON.stringify(error.actualValue);
+       let sanitisedExpected = JSON.stringify(error.expectedValue);
+
+       this._writeFailure(sanitisedMessage, sanitisedActual, sanitisedExpected);
+
+   }
+
+   private _writeUnhandledErrorOutput(error: Error): void {
+
+       this._writeFailure("The test threw an unhandled error.", "an unhandled error", "no unhandled errors to be thrown", error.stack);
+
+   }
+
+   private _writeFailure(message: string, actual: string, expected: string, stack?: string): void {
+
+       let output =
+           " ---\n" +
+           "   message: \"" + message + "\"\n" +
+           "   severity: fail\n" +
+           "   data:\n" +
+           "     got: " + actual + "\n" +
+           "     expect: " + expected + "\n";
+
+       if (stack) {
+           // encode the stack trace with base64 to prevent new lines from messing with the YAML
+           output = output + "     stack_base64: " + new Buffer(stack).toString("base64") + "\n";
```

Comment:
looking at YAML spec looks like we should do something like

```
data: |
   There once was a short man from Ealing
   Who got on a bus to Darjeeling
       It said on the door
       "Please don't spit on the floor"
   So he carefully spat on the ceiling
```

i.e. not encoding it and just using the pipe character to show it's a multiline stream (since YAML is supposed to be readable first and it avoids specific implementation of encoding making the tap-bark change simpler)